micro-rdk/src/common/credentials_storage.rs

```diff
@@ -102,15 +103,15 @@ impl From<CertificateResponse> for TlsCertificate {
 pub trait WifiCredentialStorage {
     type Error: Error + Debug + Into<ServerError>;
     fn has_wifi_credentials(&self) -> bool;
-    fn store_wifi_credentials(&self, creds: WifiCredentials) -> Result<(), Self::Error>;
+    fn store_wifi_credentials(&self, creds: &WifiCredentials) -> Result<(), Self::Error>;
```

Comment:
I changed some functions of the storage traits to accept reference instead of taking ownership. Otherwise we would have to clone each time until we can find a storage where to store the object . It's not dramatic so happy to revert that change if we feel like it doesn't make sense